src/connection.ts

```diff
@@ -260,9 +260,11 @@ export class Connection extends JSForceConnection {
 
   /**
    * Getter for the AuthInfo.
+   *
+   * @param decrypt Decrypt the fields
    */
-  public getAuthInfoFields(): AuthFields {
-    return this.options.authInfo.getFields();
+  public getAuthInfoFields(decrypt?: boolean): AuthFields {
```

Comment:
Add decrypt param function - default of undefined protects backward compatibility.

Reply:
We originally tried to not expose the decrypted values outside of the core library. I know we did that for `getFields` but I don't remember if we looked at other options.

Is there another option here? The answer may be no, but I just want to make sure we looked at other options. For example, maybe there is a way for user to use a protected method to get the authInfo rather than adding decrypt to another public method.

Obviously, anyone using the core library can decrypt if they really want to. The idea was to not make it super easy so consumers don't accidentally log/display/keep in memory when they didn't need to. Adding it to authInfo makes it a little easier. Adding it to the connection makes it really easy. At that point, we might as well just expose the authInfo.

Reply:
the other option would be to use AuthInfo directly where it's already exposed.